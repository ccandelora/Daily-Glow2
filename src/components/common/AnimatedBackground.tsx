import React, { useEffect, useRef } from 'react';
import { Animated, StyleSheet, View, Dimensions } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import theme from '@/constants/theme';

interface AnimatedBackgroundProps {
  intensity?: 'light' | 'medium' | 'strong';
  animated?: boolean;
}

export const AnimatedBackground: React.FC<AnimatedBackgroundProps> = ({ 
  intensity = 'medium',
  animated = true 
}) => {
  const fadeAnim = useRef(new Animated.Value(0)).current;
  const rotateAnim = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    if (animated) {
      // Fade in
      Animated.timing(fadeAnim, {
        toValue: 1,
        duration: 1000,
        useNativeDriver: true,
      }).start();

      // Faster rotation
      const rotateAnimation = Animated.loop(
        Animated.sequence([
          Animated.timing(rotateAnim, {
            toValue: 1,
            duration: 30000, // 30 seconds per rotation
            useNativeDriver: true,
          }),
          Animated.timing(rotateAnim, {
            toValue: 0,
            duration: 0,
            useNativeDriver: true,
          })
        ])
      );

      rotateAnimation.start();
    } else {
      fadeAnim.setValue(1);
    }
  }, [animated]);

  const spin = rotateAnim.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg']
  });

  return (
    <View style={styles.container}>
      {/* Static base gradient */}
      <LinearGradient
        colors={[
          'rgba(255,255,255,0.92)',
          'rgba(235,230,255,0.95)',
          'rgba(255,255,255,0.92)',
        ]}
        style={styles.baseGradient}
        start={{ x: 0, y: 0 }}
        end={{ x: 0, y: 1 }}
      />

      <Animated.View
        style={[
          styles.gradientContainer,
          {
            opacity: fadeAnim,
            transform: [{ rotate: spin }]
          }
        ]}
      >
        {/* Brighter purple glow */}
        <LinearGradient
          colors={[
            'transparent',
            `${theme.COLORS.primary.blue}15`,
            `${theme.COLORS.primary.blue}20`,
            `${theme.COLORS.primary.blue}15`,
            'transparent',
          ]}
          style={styles.gradient}
          start={{ x: 0.3, y: 0.3 }}
          end={{ x: 0.7, y: 0.7 }}
          locations={[0, 0.3, 0.5, 0.7, 1]}
        />

        {/* Stronger accent highlights */}
        <LinearGradient
          colors={[
            'transparent',
            'rgba(230,230,255,0.35)',
            'transparent'
          ]}
          style={styles.gradient}
          start={{ x: 0, y: 0.5 }}
          end={{ x: 1, y: 0.5 }}
        />

        {/* Enhanced color variation */}
        <LinearGradient
          colors={[
            'transparent',
            `${theme.COLORS.primary.blue}12`,
            'transparent'
          ]}
          style={styles.gradient}
          start={{ x: 0.5, y: 0 }}
          end={{ x: 0.5, y: 1 }}
        />
      </Animated.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    backgroundColor: 'white',
  },
  baseGradient: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  gradientContainer: {
    position: 'absolute',
    top: -400,
    left: -400,
    right: -400,
    bottom: -400,
  },
  gradient: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    transform: [{ scale: 2 }],
  },
}); 