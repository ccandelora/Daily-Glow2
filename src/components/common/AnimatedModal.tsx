import React, { useEffect } from 'react';
import {
  View,
  Modal,
  StyleSheet,
  TouchableOpacity,
  Animated,
  Dimensions,
  ViewStyle,
} from 'react-native';
import theme from '../../constants/theme';
import { Typography } from './Typography';

interface AnimatedModalProps {
  visible: boolean;
  onClose: () => void;
  title?: string;
  children: React.ReactNode;
  style?: ViewStyle;
}

export const AnimatedModal: React.FC<AnimatedModalProps> = ({
  visible,
  onClose,
  title,
  children,
  style,
}) => {
  const slideAnim = React.useRef(new Animated.Value(0)).current;
  const fadeAnim = React.useRef(new Animated.Value(0)).current;

  useEffect(() => {
    if (visible) {
      Animated.parallel([
        Animated.timing(slideAnim, {
          toValue: 1,
          duration: 300,
          useNativeDriver: true,
        }),
        Animated.timing(fadeAnim, {
          toValue: 1,
          duration: 300,
          useNativeDriver: true,
        }),
      ]).start();
    } else {
      Animated.parallel([
        Animated.timing(slideAnim, {
          toValue: 0,
          duration: 200,
          useNativeDriver: true,
        }),
        Animated.timing(fadeAnim, {
          toValue: 0,
          duration: 200,
          useNativeDriver: true,
        }),
      ]).start();
    }
  }, [visible, slideAnim, fadeAnim]);

  const { height } = Dimensions.get('window');

  return (
    <Modal
      transparent
      visible={visible}
      onRequestClose={onClose}
      animationType="none"
    >
      <Animated.View
        style={[
          styles.overlay,
          {
            opacity: fadeAnim,
          },
        ]}
      >
        <TouchableOpacity
          style={styles.backdrop}
          onPress={onClose}
          activeOpacity={1}
        />
        <Animated.View
          style={[
            styles.content,
            {
              transform: [
                {
                  translateY: slideAnim.interpolate({
                    inputRange: [0, 1],
                    outputRange: [height, 0],
                  }),
                },
              ],
            },
            style,
          ]}
        >
          {title && (
            <View style={styles.header}>
              <Typography variant="h3">{title}</Typography>
              <TouchableOpacity onPress={onClose} style={styles.closeButton}>
                <Typography variant="h3">Ã—</Typography>
              </TouchableOpacity>
            </View>
          )}
          {children}
        </Animated.View>
      </Animated.View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  overlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    justifyContent: 'flex-end',
  },
  backdrop: {
    ...StyleSheet.absoluteFillObject,
  },
  content: {
    backgroundColor: theme.COLORS.ui.background,
    borderTopLeftRadius: theme.BORDER_RADIUS.xl,
    borderTopRightRadius: theme.BORDER_RADIUS.xl,
    padding: theme.SPACING.lg,
    maxHeight: '90%',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: theme.SPACING.md,
  },
  closeButton: {
    padding: theme.SPACING.sm,
    margin: -theme.SPACING.sm,
  },
}); 