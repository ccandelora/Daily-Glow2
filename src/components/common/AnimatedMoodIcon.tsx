import React, { useEffect } from 'react';
import { Animated, StyleSheet, ViewStyle } from 'react-native';
import theme from '../../constants/theme';

interface AnimatedMoodIconProps {
  size?: number;
  color: string;
  style?: ViewStyle;
  children: React.ReactNode;
  active?: boolean;
}

export const AnimatedMoodIcon: React.FC<AnimatedMoodIconProps> = ({
  size = 60,
  color,
  style,
  children,
  active = false,
}) => {
  const scaleAnim = React.useRef(new Animated.Value(1)).current;
  const opacityAnim = React.useRef(new Animated.Value(0.6)).current;

  useEffect(() => {
    Animated.parallel([
      Animated.spring(scaleAnim, {
        toValue: active ? 1.1 : 1,
        useNativeDriver: true,
        damping: 10,
      }),
      Animated.timing(opacityAnim, {
        toValue: active ? 1 : 0.6,
        duration: 200,
        useNativeDriver: true,
      }),
    ]).start();
  }, [active, scaleAnim, opacityAnim]);

  return (
    <Animated.View
      style={[
        styles.container,
        {
          width: size,
          height: size,
          backgroundColor: color,
          transform: [{ scale: scaleAnim }],
          opacity: opacityAnim,
        },
        style,
      ]}
    >
      {children}
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: theme.BORDER_RADIUS.circle,
    alignItems: 'center',
    justifyContent: 'center',
    overflow: 'hidden',
  },
}); 