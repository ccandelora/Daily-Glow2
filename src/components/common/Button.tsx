import React from 'react';
import { TouchableOpacity, StyleSheet, ViewStyle, TextStyle, ActivityIndicator, StyleProp } from 'react-native';
import { Typography } from './Typography';
import theme from '@/constants/theme';

interface ButtonProps {
  title: string;
  onPress: () => void;
  variant?: 'primary' | 'secondary' | 'outline';
  size?: 'small' | 'medium' | 'large';
  disabled?: boolean;
  loading?: boolean;
  style?: ViewStyle;
  textStyle?: TextStyle;
}

export const Button: React.FC<ButtonProps> = ({
  title,
  onPress,
  variant = 'primary',
  size = 'medium',
  disabled = false,
  loading = false,
  style,
  textStyle,
}) => {
  return (
    <TouchableOpacity
      style={[
        styles.base,
        styles[variant],
        styles[size],
        disabled && styles.disabled,
        style,
      ]}
      onPress={onPress}
      disabled={disabled || loading}
    >
      {loading ? (
        <ActivityIndicator
          color={variant === 'outline' ? theme.COLORS.primary.green : theme.COLORS.ui.background}
          size="small"
        />
      ) : (
        <Typography
          variant="button"
          style={[
            styles.text,
            styles[`${variant}Text` as keyof typeof styles] as TextStyle,
            disabled && styles.disabledText,
            textStyle,
          ].filter(Boolean) as TextStyle}
        >
          {title}
        </Typography>
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  base: {
    borderRadius: theme.BORDER_RADIUS.lg,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    shadowColor: theme.COLORS.ui.accent,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 8,
    elevation: 4,
  },
  primary: {
    backgroundColor: theme.COLORS.ui.accent,
    borderWidth: 0,
    shadowColor: theme.COLORS.ui.accent,
    shadowOpacity: 0.5,
    shadowRadius: 12,
  },
  secondary: {
    backgroundColor: 'rgba(65, 105, 225, 0.1)',
    borderWidth: 1,
    borderColor: theme.COLORS.ui.accent,
  },
  outline: {
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderColor: theme.COLORS.ui.border,
  },
  small: {
    paddingVertical: theme.SPACING.sm,
    paddingHorizontal: theme.SPACING.lg,
    minWidth: 100,
  },
  medium: {
    paddingVertical: theme.SPACING.md,
    paddingHorizontal: theme.SPACING.xl,
    minWidth: 150,
  },
  large: {
    paddingVertical: theme.SPACING.lg,
    paddingHorizontal: theme.SPACING.xl * 1.5,
    minWidth: 200,
  },
  disabled: {
    backgroundColor: theme.COLORS.ui.disabled,
    borderColor: 'transparent',
    shadowOpacity: 0,
  },
  text: {
    textAlign: 'center',
  },
  primaryText: {
    color: theme.COLORS.ui.text,
    textShadowColor: 'rgba(0, 0, 0, 0.2)',
    textShadowOffset: { width: 0, height: 1 },
    textShadowRadius: 2,
  },
  secondaryText: {
    color: theme.COLORS.ui.accent,
    textShadowColor: theme.COLORS.ui.accent,
    textShadowOffset: { width: 0, height: 0 },
    textShadowRadius: 4,
  },
  outlineText: {
    color: theme.COLORS.ui.text,
  },
  disabledText: {
    color: theme.COLORS.ui.textSecondary,
  },
}); 