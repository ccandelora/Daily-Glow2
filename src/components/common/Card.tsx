import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import theme from '../../constants/theme';

interface CardProps {
  children: React.ReactNode;
  style?: ViewStyle;
  variant?: 'default' | 'elevated';
}

export const Card: React.FC<CardProps> = ({
  children,
  style,
  variant = 'default',
}) => {
  return (
    <View
      style={[
        styles.card,
        variant === 'elevated' && styles.elevated,
        style,
      ]}
    >
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: theme.COLORS.ui.card,
    borderRadius: theme.BORDER_RADIUS.lg,
    padding: theme.SPACING.md,
    marginVertical: theme.SPACING.sm,
  },
  elevated: {
    shadowColor: theme.COLORS.ui.text,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
}); 