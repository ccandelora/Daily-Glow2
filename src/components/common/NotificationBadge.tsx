import React, { useEffect } from 'react';
import { View, StyleSheet, Animated, TouchableOpacity } from 'react-native';
import { Typography } from './Typography';
import { useNotifications } from '@/contexts/NotificationsContext';
import theme from '@/constants/theme';
import { Ionicons } from '@expo/vector-icons';

interface NotificationBadgeProps {
  size?: number;
  color?: string;
  onPress?: () => void;
}

export const NotificationBadge: React.FC<NotificationBadgeProps> = ({
  size = 24,
  color = theme.COLORS.ui.text,
  onPress,
}) => {
  const { unreadCount } = useNotifications();
  const scaleAnim = React.useRef(new Animated.Value(0)).current;

  useEffect(() => {
    if (unreadCount > 0) {
      Animated.sequence([
        Animated.timing(scaleAnim, {
          toValue: 1.2,
          duration: 200,
          useNativeDriver: true,
        }),
        Animated.spring(scaleAnim, {
          toValue: 1,
          friction: 3,
          useNativeDriver: true,
        }),
      ]).start();
    } else {
      Animated.timing(scaleAnim, {
        toValue: 0,
        duration: 150,
        useNativeDriver: true,
      }).start();
    }
  }, [unreadCount]);

  const Container = onPress ? TouchableOpacity : View;

  return (
    <Container onPress={onPress} style={styles.container}>
      <Ionicons
        name="notifications-outline"
        size={size}
        color={color}
      />
      {unreadCount > 0 && (
        <Animated.View
          style={[
            styles.badge,
            {
              transform: [{ scale: scaleAnim }],
            },
          ]}
        >
          <Typography
            variant="caption"
            style={styles.badgeText}
          >
            {unreadCount > 99 ? '99+' : unreadCount}
          </Typography>
        </Animated.View>
      )}
    </Container>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'relative',
    alignItems: 'center',
    justifyContent: 'center',
  },
  badge: {
    position: 'absolute',
    top: -5,
    right: -8,
    minWidth: 18,
    height: 18,
    borderRadius: 9,
    backgroundColor: theme.COLORS.primary.red,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 4,
    borderWidth: 1.5,
    borderColor: theme.COLORS.ui.background,
  },
  badgeText: {
    color: theme.COLORS.ui.background,
    fontSize: 10,
    fontWeight: 'bold',
  },
}); 