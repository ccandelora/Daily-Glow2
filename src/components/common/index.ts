export * from './Button';
export * from './Card';
export * from './Typography';
export * from './Input';
export * from './ProgressBar';
export * from './Header';
export * from './AnimatedModal';
export * from './AnimatedMoodIcon';
export * from './EmotionWheel';
export { LoadingSpinner } from './LoadingSpinner';
export { Toast } from './Toast';
export { LoadingOverlay } from './LoadingOverlay';
export { DailyChallenge } from './DailyChallenge';
export { NotificationBadge } from './NotificationBadge';
export { TabBar } from './TabBar'; 