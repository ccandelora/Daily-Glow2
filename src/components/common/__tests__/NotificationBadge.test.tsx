import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { NotificationBadge } from '@/components/common/NotificationBadge';
import { useNotifications } from '@/contexts/NotificationsContext';
import { TouchableOpacity, View, Animated } from 'react-native';

// Mock the NotificationsContext
jest.mock('@/contexts/NotificationsContext', () => ({
  useNotifications: jest.fn()
}));

// Mock FontAwesome6 from @expo/vector-icons
jest.mock('@expo/vector-icons', () => ({
  FontAwesome6: 'FontAwesome6'
}));

describe('NotificationBadge Component', () => {
  // Set up mocks for different scenarios
  const mockOnPress = jest.fn();
  
  beforeEach(() => {
    jest.clearAllMocks();
  });
  
  it('renders correctly with zero unread notifications', () => {
    // Mock the useNotifications hook for 0 unread count
    (useNotifications as jest.Mock).mockReturnValue({ unreadCount: 0 });
    
    const { toJSON, queryByText } = render(
      <NotificationBadge />
    );
    
    // Badge should not be visible when unreadCount is 0
    expect(queryByText('0')).toBeFalsy();
    
    // Should match snapshot
    expect(toJSON()).toMatchSnapshot();
  });
  
  it('renders correctly with unread notifications', () => {
    // Mock the useNotifications hook for 5 unread count
    (useNotifications as jest.Mock).mockReturnValue({ unreadCount: 5 });
    
    const { toJSON, getByText } = render(
      <NotificationBadge />
    );
    
    // Badge should be visible and show the correct count
    expect(getByText('5')).toBeTruthy();
    
    // Should match snapshot
    expect(toJSON()).toMatchSnapshot();
  });
  
  it('renders 99+ when unread notifications exceed 99', () => {
    // Mock the useNotifications hook for 100 unread count
    (useNotifications as jest.Mock).mockReturnValue({ unreadCount: 100 });
    
    const { toJSON, getByText } = render(
      <NotificationBadge />
    );
    
    // Badge should show 99+ instead of the actual value
    expect(getByText('99+')).toBeTruthy();
    
    // Should match snapshot
    expect(toJSON()).toMatchSnapshot();
  });
  
  it('renders as TouchableOpacity when onPress is provided', () => {
    // Mock the useNotifications hook
    (useNotifications as jest.Mock).mockReturnValue({ unreadCount: 5 });
    
    const { UNSAFE_getAllByType } = render(
      <NotificationBadge onPress={mockOnPress} />
    );
    
    // Should render as TouchableOpacity
    const touchables = UNSAFE_getAllByType(TouchableOpacity);
    expect(touchables.length).toBeGreaterThan(0);
  });
  
  it('renders as View when onPress is not provided', () => {
    // Mock the useNotifications hook
    (useNotifications as jest.Mock).mockReturnValue({ unreadCount: 5 });
    
    const { UNSAFE_getAllByType } = render(
      <NotificationBadge />
    );
    
    // Should render as View
    const views = UNSAFE_getAllByType(View);
    expect(views.length).toBeGreaterThan(0);
  });
  
  it('calls onPress when pressed', () => {
    // Mock the useNotifications hook
    (useNotifications as jest.Mock).mockReturnValue({ unreadCount: 5 });
    
    const { UNSAFE_getAllByType } = render(
      <NotificationBadge onPress={mockOnPress} />
    );
    
    // Get the TouchableOpacity and simulate press
    const touchable = UNSAFE_getAllByType(TouchableOpacity)[0];
    fireEvent.press(touchable);
    
    // onPress should be called
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
  
  it('applies custom size prop', () => {
    // Mock the useNotifications hook
    (useNotifications as jest.Mock).mockReturnValue({ unreadCount: 5 });
    
    const { toJSON } = render(
      <NotificationBadge size={32} />
    );
    
    // Should match snapshot with custom size
    expect(toJSON()).toMatchSnapshot();
  });
  
  it('applies custom color prop', () => {
    // Mock the useNotifications hook
    (useNotifications as jest.Mock).mockReturnValue({ unreadCount: 5 });
    
    const { toJSON } = render(
      <NotificationBadge color="red" />
    );
    
    // Should match snapshot with custom color
    expect(toJSON()).toMatchSnapshot();
  });
}); 