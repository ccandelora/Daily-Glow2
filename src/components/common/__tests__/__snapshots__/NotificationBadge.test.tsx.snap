// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBadge Component applies custom color prop 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "position": "relative",
    }
  }
>
  <FontAwesome6
    color="red"
    name="bell"
    size={24}
  />
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF6B6B",
        "borderColor": "rgba(28, 14, 45, 0.95)",
        "borderRadius": 9,
        "borderWidth": 1.5,
        "height": 18,
        "justifyContent": "center",
        "minWidth": 18,
        "paddingHorizontal": 4,
        "position": "absolute",
        "right": -8,
        "top": -5,
        "transform": [
          {
            "scale": 0,
          },
        ],
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 16,
          },
          {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 14,
            "fontWeight": "400",
          },
          undefined,
          null,
          {
            "color": "rgba(28, 14, 45, 0.95)",
            "fontSize": 10,
            "fontWeight": "bold",
          },
        ]
      }
    >
      5
    </Text>
  </View>
</View>
`;

exports[`NotificationBadge Component applies custom size prop 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "position": "relative",
    }
  }
>
  <FontAwesome6
    color="#FFFFFF"
    name="bell"
    size={32}
  />
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF6B6B",
        "borderColor": "rgba(28, 14, 45, 0.95)",
        "borderRadius": 9,
        "borderWidth": 1.5,
        "height": 18,
        "justifyContent": "center",
        "minWidth": 18,
        "paddingHorizontal": 4,
        "position": "absolute",
        "right": -8,
        "top": -5,
        "transform": [
          {
            "scale": 0,
          },
        ],
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 16,
          },
          {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 14,
            "fontWeight": "400",
          },
          undefined,
          null,
          {
            "color": "rgba(28, 14, 45, 0.95)",
            "fontSize": 10,
            "fontWeight": "bold",
          },
        ]
      }
    >
      5
    </Text>
  </View>
</View>
`;

exports[`NotificationBadge Component renders 99+ when unread notifications exceed 99 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "position": "relative",
    }
  }
>
  <FontAwesome6
    color="#FFFFFF"
    name="bell"
    size={24}
  />
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF6B6B",
        "borderColor": "rgba(28, 14, 45, 0.95)",
        "borderRadius": 9,
        "borderWidth": 1.5,
        "height": 18,
        "justifyContent": "center",
        "minWidth": 18,
        "paddingHorizontal": 4,
        "position": "absolute",
        "right": -8,
        "top": -5,
        "transform": [
          {
            "scale": 0,
          },
        ],
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 16,
          },
          {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 14,
            "fontWeight": "400",
          },
          undefined,
          null,
          {
            "color": "rgba(28, 14, 45, 0.95)",
            "fontSize": 10,
            "fontWeight": "bold",
          },
        ]
      }
    >
      99+
    </Text>
  </View>
</View>
`;

exports[`NotificationBadge Component renders correctly with unread notifications 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "position": "relative",
    }
  }
>
  <FontAwesome6
    color="#FFFFFF"
    name="bell"
    size={24}
  />
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF6B6B",
        "borderColor": "rgba(28, 14, 45, 0.95)",
        "borderRadius": 9,
        "borderWidth": 1.5,
        "height": 18,
        "justifyContent": "center",
        "minWidth": 18,
        "paddingHorizontal": 4,
        "position": "absolute",
        "right": -8,
        "top": -5,
        "transform": [
          {
            "scale": 0,
          },
        ],
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 16,
          },
          {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 14,
            "fontWeight": "400",
          },
          undefined,
          null,
          {
            "color": "rgba(28, 14, 45, 0.95)",
            "fontSize": 10,
            "fontWeight": "bold",
          },
        ]
      }
    >
      5
    </Text>
  </View>
</View>
`;

exports[`NotificationBadge Component renders correctly with zero unread notifications 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "position": "relative",
    }
  }
>
  <FontAwesome6
    color="#FFFFFF"
    name="bell"
    size={24}
  />
</View>
`;
